clients/drcachesim/simulator/snoop_filter.cpp

```diff
@@ -74,14 +74,8 @@ void
 snoop_filter_t::snoop(addr_t tag, int id, bool is_write)
 {
     coherence_table_entry_t *coherence_entry = &coherence_table_[tag];
-    // Initialize new snoop filter entry.
-    if (coherence_entry->sharers.empty()) {
-        coherence_entry->sharers.resize(num_snooped_caches_, false);
-        coherence_entry->dirty = false;
-    }
 
-    auto num_sharers = std::count(coherence_entry->sharers.begin(),
-                                  coherence_entry->sharers.end(), true);
+    auto num_sharers = coherence_entry->sharers.size();
```

Comment:
s/auto/int/

Better to avoid auto when it doesn't improve readability.

Reply:
As I recall my syntax checker had thoughts on the proper type and flagged int with a warning, so I punted and just went with auto.  But it seems other code here either casts the size() value to int (not helping readability!), or else uses size_t.  So size_t it is.